src/lightning_app/core/app.py

```diff
@@ -568,6 +653,32 @@ def _dump_checkpoint(self) -> Optional[str]:
             pickle.dump(self.state_dict(), f)
         return checkpoint_path
 
+    def _save_checkpoint(self, checkpoint_name: str = "") -> Optional[str]:
+        """Save the current state of the app in a checkpoint. If the app is running locally, the checkpoint will be
+        saved in the local directory. If the app is running on the cloud, the checkpoint will be saved in the cloud
+        storage.
+
+        :param checkpoint_name: The name of the checkpoint, if not defined, the checkpoint will be saved with the
+            current timestamp.
+        :return: the path of the checkpoint file.
+        """
+
+        if not checkpoint_name:
+            checkpoint_name = f"lightningapp_checkpoint_{int(time())}"
+        os.makedirs(self.checkpoint_dir, exist_ok=True)
+        checkpoint_path = os.path.join(self.checkpoint_dir, f"{checkpoint_name}.json")
+
+        with open(checkpoint_path, "w") as f:
+            state = apply_to_collection(self.state, (Path, Drive), lambda x: x.to_dict())
+            json.dump(state, f)
+
+        drive = Drive(
+            "lit://checkpoints", component_name="root", allow_duplicates=True, root_folder=self.checkpoint_dir
+        )
+        with _context(ComponentContext.WORK):
```

Comment:
Why are we pretending to be in a work context? Let's not do this. Can't we just save it directly to the shared path?

Reply:
That's also my question with these work contexts, can you clarify what is the underlying holdup @manskx ?

Reply:
It's because the drive doesn't allow the flow to put/get/list files 
https://github.com/Lightning-AI/lightning/blob/e9a6b834370251252d0e6c9767cac8bd7c81407b/src/lightning_app/storage/drive.py#L90

Reply:
@tchaton what is the best practice here?